### PR TITLE
Fixes #239: Add options consistency check for 

### LINKED ISSUE #239: Handle options consistency issues
Throw error if none of `labels` `query` or `relationship` is set.

Log a warning for other kind of inconsistent options.

### PR DESCRIPTION
**Pull request by utnaf**
Fixes #239 

I think giving a warning for the ignored options is better that breaking the execution, but let me know.

### CODE REVIEW

File: src/main/scala/org/neo4j/spark/util/Validations.scala [hunk 1]
```diff
@@ -134,6 +141,43 @@ object Validations {
     }
   }
 
+  def checkOptionsConsistency(neo4jOptions: Neo4jOptions): Unit = {
```
Comment (conker84) [inline src/main/scala/org/neo4j/spark/util/Validations.scala:150]: can you add just a few lines for describing what this method does internally? It took a while to me to understand the behaviour

File: src/main/scala/org/neo4j/spark/Neo4jOptions.scala [hunk 2]
```diff
@@ -48,12 +48,7 @@ class Neo4jOptions(private val parameters: java.util.Map[String, String]) extend
       Neo4jQueryOptions(LABELS, parsed)
     }
     case ("", "", relationship) => Neo4jQueryOptions(RELATIONSHIP, relationship)
-    case _ => throw new IllegalArgumentException(
-      s"You need to specify just one of these options: ${
-        QueryType.values.toSeq.map(value => s"'${value.toString.toLowerCase()}'")
-          .sorted.mkString(", ")
-      }"
-    )
+    case _ => throw new IllegalArgumentException("You must specify one of `query`, `labels`, `relationship`")
```
Comment (conker84) [inline src/main/scala/org/neo4j/spark/Neo4jOptions.scala]: why this change?

File: src/test/scala/org/neo4j/spark/Neo4jOptionsTest.scala [hunk 3]
```diff
@@ -42,45 +42,6 @@ class Neo4jOptionsTest {
     assertEquals(NodeSaveMode.Overwrite, neo4jOptions.relationshipMetadata.targetSaveMode)
   }
 
-  @Test
```
Comment (conker84) [inline src/test/scala/org/neo4j/spark/Neo4jOptionsTest.scala:45]: why you removed these?

**Review by conker84 — COMMENTED** [review]
thank you @utnaf can you please look at my questions?

File: src/test/scala/org/neo4j/spark/Neo4jOptionsTest.scala (hunk 3 above)
Comment (utnaf) [reply src/test/scala/org/neo4j/spark/Neo4jOptionsTest.scala:45]: Because I added the same tests on the Writer and Reader IT

File: src/main/scala/org/neo4j/spark/Neo4jOptions.scala (hunk 2 above)
Comment (utnaf) [reply src/main/scala/org/neo4j/spark/Neo4jOptions.scala]: This was unwanted. I'll restore it. It doesn't change much much but I prefer the previous one

**Review by conker84 — APPROVED** [review]

**Comment by conker84** [comment]
@moxious I'm merging this. @utnaf thank you
